src/psyclone/psyir/transformations/omp_target_trans.py

```diff
@@ -158,11 +158,12 @@ def validate(self, node, options=None):
         :raises TransformationError: if its a function and the target region
             attempts to enclose the assingment setting the return value.
         '''
+        device_string = options.get("device_string", "") if options else ""
         node_list = self.get_node_list(node)
         super().validate(node, options)
         for node in node_list:
             for call in node.walk(Call):
-                if not call.is_available_on_device():
+                if not call.is_available_on_device(device_string):
                     raise TransformationError(
                         f"'{call.routine.name}' is not available on the "
                         f"accelerator device, and therefore it cannot "
```

Comment:
Include `device_string` in this message, e.g. "with device 'nvidia-repr'"? We should perhaps s/nvidia-repr/nvidia-reproducible/ in the interests of readability?

Reply:
I agree with the readability, but I wasn't happy about calling it reproducible because it is not quite right (nvfortran-all is reproducible between GPU runs, the nvfortran-repr was not reproducible unless using math_uniform flag, it doesn't guarantee reproducibility if there is atomics/reductions). So, I changed it to "nvfortran-uniform" to use the same semantic as the compiler and hopefully makes clearer that it needs to be combined with math_uniform.

Reply:
Done